internal/frr/api_to_config.go

```diff
@@ -0,0 +1,74 @@
+package frr
+
+import (
+	"fmt"
+
+	v1beta1 "github.com/metallb/frrk8s/api/v1beta1"
+	"github.com/metallb/frrk8s/internal/ipfamily"
+)
+
+func routerToFRRConfig(r v1beta1.Router) (*routerConfig, error) {
+	res := &routerConfig{
+		MyASN:        r.ASN,
+		RouterID:     r.ID,
+		VRF:          r.VRF,
+		Neighbors:    make([]*neighborConfig, 0),
+		IPV4Prefixes: make([]string, 0),
+		IPV6Prefixes: make([]string, 0),
+	}
+
+	for _, n := range r.Neighbors {
+		frrNeigh, err := neighborToFRR(n)
+		if err != nil {
+			return nil, err
+		}
+		res.Neighbors = append(res.Neighbors, frrNeigh)
+	}
+	for _, p := range r.Prefixes {
+		family := ipfamily.ForCIDRString(p)
+		switch family {
+		case ipfamily.IPv4:
+			res.IPV4Prefixes = append(res.IPV4Prefixes, p)
+		case ipfamily.IPv6:
+			res.IPV6Prefixes = append(res.IPV6Prefixes, p)
+		case ipfamily.Unknown:
+			return nil, fmt.Errorf("unknown ipfamily for %s", p)
+		}
+
+	}
+	return res, nil
+}
+
+func neighborToFRR(n v1beta1.Neighbor) (*neighborConfig, error) {
+	neighborFamily, err := ipfamily.ForAddresses(n.Address)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find ipfamily for %s, %w", n.Address, err)
+	}
+	res := &neighborConfig{
+		Name: neighborName(n.ASN, n.Address),
+		ASN:  n.ASN,
+		Addr: n.Address,
+		Port: n.Port,
+		// Password:       n.Password, TODO password as secret
+		Advertisements: make([]*advertisementConfig, 0),
+		IPFamily:       neighborFamily,
+		EBGPMultiHop:   n.EBGPMultiHop,
+	}
+	// TODO allow all
+	for _, p := range n.ToAdvertise.Allowed.Prefixes {
```

Comment:
Yep, there's a lot of validation that is not currently done and we might want to do. As above, I'd rather cover with tests and implement instead of just implement with no tests (and I don't want this to be gigantic)